src/library/scala/collection/mutable/FlatHashTable.scala

```diff
@@ -415,20 +418,7 @@ private[collection] object FlatHashTable {
     // so that:
     protected final def sizeMapBucketSize = 1 << sizeMapBucketBitSize
 
-    protected final def improve(hcode: Int, seed: Int) = {
-      //var h: Int = hcode + ~(hcode << 9)
-      //h = h ^ (h >>> 14)
-      //h = h + (h << 4)
-      //h ^ (h >>> 10)
-
-      val improved= scala.util.hashing.byteswap32(hcode)
-
-      // for the remainder, see SI-5293
-      // to ensure that different bits are used for different hash tables, we have to rotate based on the seed
-      val rotation = seed % 32
-      val rotated = (improved >>> rotation) | (improved << (32 - rotation))
-      rotated
-    }
+    protected final def improve(hcode: Int, seed: Int) = rotateRight(byteswap32(hcode), seed & 31)
```

Comment:
From [the Scaladoc](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#rotateRight(int,%20int)), `rotateRight(val, distance) == rotateRight(val, distance & 0x1F)`, so the `& 31` is redundant here, and this can be simplified as
```scala
rotateRight(byteswap32(hcode), seed)
```